mdtraj/core/topology.py

```diff
@@ -1621,14 +1621,10 @@ def __eq__(self, other):
             return False
         if {other[0], other[1]} != {self[0], self[1]}:
             return False
-        # Optional consideration, but this is more like metadata on the bond,
-        # If the above checks out, but the below does not, then there are
-        # now two bonds between the same atoms, which makes no sense
-
-        # if other.type != self.type:
-        #     return False
-        # if other.order != self.order:
-        #     return False
+        if other.type != self.type:
+            return False
+        if other.order != self.order:
+            return False
```

Comment:
I don't think you have a total ordering now, since you changed eq w/o changing gt and lt. two bonds that share the same atoms but have different type or order are neither greater than, less than, nor equal to one another under the current code.)

Reply:
Under that condition (same atom indices, but different type/order), that should probably throw an error somewhere since that means there are two `Bonds` between the same pair of atoms (as you mentioned before). I can't think of a use case where that may be helpful, unless you are defining a type of additional restraint through the "bond" framework of whatever simulation package you are using.

 I'm not sure where there would be a good time to check for multiple bonds on the same atoms. The `add_bond` method could check to see if there is already a bond between atom1 and atom2 in the Topology._bonds list, but will that add too much overhead? The only way you could have 2 bonds on the same pair of atoms is if you manually call `add_bond` right now; I don't think any of the programatic ways (e.g. `from_openmm`, `join`) can cause this case. 

If you don't mind the overhead of doing a `for a,b in self._bonds` each time `add_bond` is called, I can add this in and then not have to worry about the inequality. Alternately, I can add an absolute order to `order` and `type` properties for `Bond.__lt__` and `Bond.__gt__`.

Reply:
As I said before, I'm really fine with *either* it being illegal to have multiple bonds between the same pair of atoms or comparison operators that properly handle multiple bonds between the same pair of atoms. I wouldn't worry about the overhead of checking -- it's easy to optimize if slow.

Reply:
you ought not to have to write _all_ of the comparators either -- [@functools.total_ordering](https://docs.python.org/3/library/functools.html#functools.total_ordering) can fill in most of them